msgraph/client.go

```diff
@@ -126,6 +141,14 @@ func (c Client) performRequest(req *http.Request, input HttpRequestInput) (*http
 		return base * backoffPower(base, exp-1)
 	}
 
+	var reqBody []byte
+	if req.Body != nil {
+		reqBody, err = io.ReadAll(req.Body)
+		if err != nil {
+			return nil, status, nil, fmt.Errorf("reading request body: %v", err)
+		}
+	}
+
 	var attempts, backoff, multiplier int64
 	for attempts = 0; attempts < requestAttempts; attempts++ {
```

Comment:
It is slightly hard to understand for how long the retries will be performed. 
Correct me if I am wrong: 
- we have `10 attempts`.
- every time we have a consistency error, the `backoff` is set to `2`.
- in between attempts, we sleep for `2` seconds.

i.e., at most, we are retrying for ~20 seconds, right ?

***
Some time ago, I stumbled upon [this](https://youtu.be/AeLDtcpKc-U?t=418) video, where the author talks about `1-2 minute` for replication to propagate fully. 

We are daily provisioning applications and service principals, and I can say statistically it is about 2 minutes. i.e., we stopped receiving errors when we increased retries to 2 minutes mark.

I am afraid that ~20 seconds won't be enough. Or, we can start with something (`20 seconds`) and see how it goes.